autoload/beancount.vim

```diff
@@ -56,14 +66,29 @@ function! beancount#complete_account(findstart, base)
         endif
         let b:beancount_accounts = beancount#find_accounts(l:root)
     endif
-    let l:pattern = '\V' . substitute(a:base, ":", '\\S\\*:\\S\\*', "g")
+
+    if g:beancount_account_completion ==? 'chunks'
+        let l:pattern = '\V' . substitute(a:base, ":", '\\[^ \\t:]\\*:', "g") . '\[^ \t:]\*'
```

Comment:
I now see that this pattern is dual purpose, used to extract match substring in addition to determining match. But this raises another question: won't all matches have the exact same number of parts? i.e. g:beancount_detailed_first is irrelevant if g:beancount_account_completion is 'chunks'?